autosklearn/data/feature_validator.py

```diff
@@ -0,0 +1,399 @@
+import functools
+import typing
+import warnings
+
+import numpy as np
+
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+
+import scipy.sparse
+
+import sklearn.utils
+from sklearn import preprocessing
+from sklearn.base import BaseEstimator
+from sklearn.compose import make_column_transformer
+
+
+class FeatureValidator(BaseEstimator):
+
+    def __init__(self) -> None:
+        # If a dataframe was provided, we populate
+        # this attribute with the column types from the dataframe
+        # That is, this attribute contains whether autosklearn
+        # should treat a column as categorical or numerical
+        # During fit, if the user provided feat_types, the user
+        # constrain is honored. If not, this attribute is used.
+        self.feature_types = None  # type: typing.Optional[typing.List[str]]
+
+        self.data_type = None  # type: typing.Optional[type]
+
+        self.encoder = None  # type: typing.Optional[BaseEstimator]
+        self.enc_columns = []  # type: typing.List[str]
+
+        self._is_fitted = False
+
+    def fit(
+        self,
+        X_train: typing.Union[np.ndarray, pd.DataFrame],
```

Comment:
Same as in the target encoder. Are lists supported? If they are, maybe we don't have to add support for them right away but it suffices to do so in a new PR?

Reply:
List are now supported!